Merge remakes and clones

tzizi opened the pull request:
Partially addresses #583. I also added a bit of wording in the index.html file to make the distinction between a remake and a clone clearer to the viewer.

nikuda commented:
I'll have to check this one out from home. Not sure if the python code is the clearest way of doing this.

nikuda commented:
Do you mind rebasing from master? It seems like a lot of the changes here have already made it into master from the other PRs you had. Thanks for the clean up by the way 👍

tzizi commented:
Sorry, I'm a bit new to this. I tried rebasing, but github's diff doesn't seem to work?

nikuda commented:
You need to be able to pull in changes from `opengaming:master` to your local repo. 

```
git remote add upstream git@github.com:opengaming/osgameclones.git
```
then this should do it

```
git fetch upstream
git rebase upstream/master combine-remakes-clones
git push
```

tzizi commented:
Thanks! Fixed :)

cxong commented on _ext.py:
```diff
@@ -182,8 +189,8 @@ def sort_key(game):
             combined['clones'] = combined_clones
-        parse_items(site, combined, 'remakes')
-        parse_items(site, combined, 'clones')
+        combined['both'] = combined_remakes + combined_clones
```
use something like 'games' instead of 'both' - after all, one day this would include more things? So 'both' would be inaccurate 😉

cxong commented on _ext.py:
```diff
@@ -182,8 +189,8 @@ def sort_key(game):
         ]
         if len(combined_clones) > 0:
             combined['clones'] = combined_clones
```
'clones' and 'remakes' aren't needed, since only 'both' is used

cxong commented on index.html:
```diff
@@ -72,19 +72,15 @@ <h1>Open Source Game Clones</h1>
+      A <span class="tag" data-name="remake">Remake</span> is a game where the executable and sometimes the assets as well are remade open source. Some of these games aren&#39;t exact remakes but evolution of original ones, which were eventually open sourced.
+      <br />
+      A <span class="tag" data-name="clone">Clone</span> is a game which is very similar to or heavily inspired by a game or series.
+    </article>
```
👍

tzizi commented:
Thanks, fixed :)

What is your point of view on using the `type` attribute I suggested in #583 ? It would make the code much cleaner I believe, something like this:

```
combined['games'] = [
    clone for clone in clones
    if name in clone['originals']
]
```

And the tags hack could be eliminated:

```
game_tags = [
    ...
    'type'
]
```

With, of course, games.html being updated as well.

nikuda commented:
There have been discussions about having the `type: clone | remake` attribute for a while now, and I'd say everyone's in agreement that's the way it should be done - but no-one has gone and done it yet. PR for it would be welcome 👍

tzizi commented:
Gone and done it :)

Any ideas for the hover background for the Remake/Clone tags?

tzizi commented:
Rebased again, the commits look fine.

However, I found a bug with the tags. When clicking on the "Clone", if you scroll down to "Microsoft Train Simulator" (just an example, there are many others), you'll find OpenRails as expected, but also ICBM3D, which is a clone of Missile Command, repeated twice (for some reason, it appears to be duplicated, also seen on the website before this PR).

I'm not sure if this is a bug intrinsic in the tag system, or if I introduced it with this PR. I'll have to look into it.

tzizi commented:
Interesting, rebased again, however now it appears the Missile Command issue still exists (although the Super Methane Brothers one is now fixed)
